.github/actions/templates/avm-validateModuleDeployment/action.yml

```diff
@@ -55,14 +55,47 @@ inputs:
   customLocation:
     description: "Custom location overwrite, if needed"
     required: false
+  e2eignore:
+    description: "A test case will not be deployed if an .e2eignore file exists"
+    required: false
 
 runs:
   using: "composite"
   steps:
+    # [Deployment validation] task(s)
+    # -------------------------------
+    - name: "Validate Test Execution"
+      uses: azure/powershell@v2
+      with:
+        azPSVersion: "latest"
+        errorActionPreference: "SilentlyContinue"
+        inlineScript: |
+          # the first step is to check, if the deployment should be bypassed
+
+          # Resolve template file path
+          $moduleTestFilePath = Join-Path $env:GITHUB_WORKSPACE '${{ inputs.templateFilePath }}'
+
+          # Check if the deployment test should be bypassed
+          if (-not [String]::IsNullOrEmpty('${{ inputs.e2eignore }}') -and '${{ inputs.e2eignore }}' -eq 'true') {
+            Write-Warning -Message "File '.e2eignore' exists in the folder: $moduleTestFilePath"
+            Write-Output 'skip_deployment_ci=true' >> $env:GITHUB_ENV
+
+            # get the reason for skipping the deployment
+            $excludeReason = Get-Content Join-Path $moduleTestFilePath '.e2eignore'
+            excludeMessage = "Skipping deployment for ${{ inputs.templateFilePath }}: $excludeReason"
+
+            Write-Output "Skipping deployment for ${{ inputs.templateFilePath }}: $excludeMessage" >> $env:VALIDATE_TEST_EXECUTION_RESULT
+            Write-Error "Skipping deployment for ${{ inputs.templateFilePath }}: $excludeMessage"
```

Comment:
Why a `Write-Error` and not e.g. `Write-Warning`? Is this supposed to be an error-state?